### PR TITLE
Todo-list exercise

### PR DESCRIPTION
**Pull request by haifeich**

### CODE REVIEW

File: 03-react/07-todo-list/src/main.tsx [hunk 1]
```diff
@@ -1,12 +1,137 @@
...
+              <input
+                type="checkbox"
+                defaultChecked={todo.done}
+                onChange={() => {
+                  todo.done = !todo.done;
+                  setTodos(
+                    replace(
+                      todos,
+                      todo,
+                      todos.findIndex((item) => item.created === todo.created)
```
Comment (mbovel) [inline 03-react/07-todo-list/src/main.tsx:112]: Why not use the index argument passed to the map lambda?

File: 03-react/07-todo-list/src/main.tsx [hunk 2]
```diff
@@ -1,12 +1,137 @@
...
+                    )
+                  );
+
+                  // need to add the new array to the localStorage, or todos?
+                  localStorage.setItem(
```
Comment (mbovel) [inline 03-react/07-todo-list/src/main.tsx:117]: The new array.

Comment (mbovel) [reply 03-react/07-todo-list/src/main.tsx:117]: Note, using an effect hook with the `todos` state variable as dependency might be more elegant, and would avoid saving the item to local storage at two different places.

**Review by mbovel — APPROVED** [review]

File: 03-react/07-todo-list/src/main.tsx (hunk 1 above)
Comment (haifeich) [reply 03-react/07-todo-list/src/main.tsx:112]: I used the index argument, but it's the map method works on the filtertodo() array, not the todo array. The index number is different.

**Pull request merged**